includes/class-wc-gateway-komoju-ipn-handler.php

```diff
@@ -113,115 +131,91 @@ protected function validate_amount( $order, $amount ) {
 	/**
 	 * Handle a captured payment
 	 * @param  WC_Order $order
+	 * @param  WC_Gateway_Komoju_Webhook_Event $webhookEvent Webhook event data
 	 */
-	protected function payment_status_captured( $order, $posted ) {
+	protected function payment_status_captured( $order, $webhookEvent ) {
 		if ( $order->has_status( 'captured' ) ) {
-			WC_Gateway_Komoju::log( 'Aborting, Order #' . $order->id . ' is already complete.' );
+			WC_Gateway_Komoju::log( 'Aborting, Order #' . $order->get_id() . ' is already complete.' );
 			exit;
 		}
 		
-		$this->validate_currency( $order, $posted['currency'] );
-		$this->validate_amount( $order, $posted['grand_total']-$posted['payment_method_fee'] ); 
-		$this->save_komoju_meta_data( $order, $posted );
+		$this->validate_currency( $order, $webhookEvent->currency() );
+		$this->validate_amount( $order, $webhookEvent->grand_total() - $webhookEvent->payment_method_fee() ); 
+		$this->save_komoju_meta_data( $order, $webhookEvent );
 
-		if ( 'captured' === $posted['status'] ) {
-			$this->payment_complete( $order, ( ! empty( $posted['external_order_num'] ) ? wc_clean( $posted['external_order_num'] ) : '' ), __( 'IPN payment captured', 'komoju-woocommerce' ) );
+		if ( 'captured' === $webhookEvent->status() ) {
+			$this->payment_complete( $order, ( ! empty( $webhookEvent->external_order_num() ) ? wc_clean( $webhookEvent->external_order_num() ) : '' ), __( 'IPN payment captured', 'komoju-woocommerce' ) );
 
-			if ( ! empty( $posted['payment_method_fee'] ) ) {
+			if ( ! empty( $webhookEvent->payment_method_fee() ) ) {
 				// log komoju transaction fee
-				update_post_meta( $order->id, 'Payment Gateway Transaction Fee', wc_clean( $posted['payment_method_fee'] ) );
+				update_post_meta( $order->get_id(), 'Payment Gateway Transaction Fee', wc_clean( $webhookEvent->payment_method_fee() ) );
 			}
 
 		} else {
-			$this->payment_on_hold( $order, sprintf( __( 'Payment pending: %s', 'woocommerce-konomu' ), $posted['additional_information'] ) );
+			$this->payment_on_hold( $order, sprintf( __( 'Payment pending: %s', 'woocommerce-konomu' ), $webhookEvent->additional_information() ) );

```

Comment:
oops, just spotted this typo: `woocommerce-konomu` -> `woocommerce-komoju` 😄. As I found out during development if the plugin name is incorrect then the language files won't be properly applied.

Will fix 👍

Reply:
Resolved in cbf23ac.